src/v/cloud_storage/partition_manifest.cc

```diff
@@ -1097,6 +1128,36 @@ void partition_manifest::serialize(std::ostream& out) const {
         }
         w.EndObject();
     };
+    auto serialize_lw_meta = [this, &w](const lw_segment_meta& meta) {
```

Comment:
Looks like there is a serialize function but not corresponding deserialization?  If the intent is that the lw_segment_meta json representation is decodeable as a segment_meta with some missing fields, then a comment here to explain that would be good.

Reply:
because lw_segment_meta is a subset of the segment_meta, the code which deserializes them is the same